WooCommerce/Classes/ViewRelated/Orders/Collect Payments/CardPresentPaymentAlertsPresenter.swift

```diff
@@ -6,6 +6,7 @@ protocol CardPresentPaymentAlertsPresenting<AlertDetails> {
     associatedtype AlertDetails
     func present(viewModel: AlertDetails)
     func presentWCSettingsWebView(adminURL: URL, completion: @escaping () -> Void)
+    func presentMerchantEducation(completion: @escaping () -> Void)
```

Comment:
❓ Just feeling a bit uncertain about this, as this is just for TTP and I'm not sure if it requires a separate function in fear of this protocol growing very long in the future with new views like this 🤔 @joshheald WDYT about having a separate function here (maybe renaming to `presentTTPMerchantEducation` since this is specific to TTP), or include this merchant education in `present(viewModel: AlertDetails)`? This education UI has a different presentation style as a fullscreen sheet though, it probably requires more work to allow a view model other than `CardPresentPaymentsModalViewModel` in the UIKit version `CardPresentPaymentAlertsPresenter`.

Reply:
Yes, I was going back and forth on how to better approach this. 

Integrating within `func present(viewModel: AlertDetails)` would go against a current setup. We want a different ViewModel and a different View for TTP education.

We could also create a separate presenter whose sole responsibility would be to present TTP education on top of the existing context.

Reply:
> Integrating within `func present(viewModel: AlertDetails)` would go against a current setup. We want a different ViewModel and a different View for TTP education.

I was thinking of refactoring the UIKit version `CardPresentPaymentAlertsPresenter` to use a generic `AlertDetails` similar to the POS version so that we can decide how to present an alert detail. But it will likely require significant changes to maintain backward compatibility.

> We could also create a separate presenter whose sole responsibility would be to present TTP education on top of the existing context. ([Example](https://github.com/woocommerce/woocommerce-ios/compare/fix/14327-ttpreq-integrate-education-steps-after-terms-and-conditions-are-accepted...fix/14327-ttpreq-integrate-education-steps-after-terms-and-conditions-are-accepted-improvement?expand=1))

Nice, I like this approach - it doesn't add complexity to the alerts presenter and the merchant education UI is not an "alert" anyway. When the time comes to support TTP in POS, we can update `BuiltInCardReaderMerchantEducationPresenter` to maybe a protocol to support both UIKit and SwiftUI. Lemme know what you think of going with this approach!

Reply:
I will start manual testing after this thread concludes.

Reply:
Yes, I think the second approach of having a separate presenter for education is reasonable. Non-TTP-related controllers don't have to have any knowledge of it. I will include it in the main PR.

Reply:
Updated in https://github.com/woocommerce/woocommerce-ios/pull/14580/commits/dab2e9d149bb458f8d6d454adc8828026162f53f

Reply:
I can see why you went back and forth here!

> I was thinking of refactoring the UIKit version CardPresentPaymentAlertsPresenter to use a generic AlertDetails similar to the POS version so that we can decide how to present an alert detail. But it will likely require significant changes to maintain backward compatibility.

In some ways it does seem a little strange _not_ to include alerts which (currently) only show during a built in reader connection in `CardPresentPaymentBuiltInReaderConnectionAlertsProvider`. However... just because they (currently) only happen during that connection doesn't make them a connection alert. The needs we have for education are different than we're showing information about an ongoing connection. It's not even quite part of the connection flow; it sort of floats above it.

It would be perfectly valid to show education outside a connection flow as well, and if we did that, it would seem strange if we _needed_ a `CardPresentPaymentEventDetails` to do it. In the new design, we'd be fine: just have some new `CardPresentPaymentEventDetails` case like `showTapToPayEducation`, which the presenter would then use to create a `TapToPayEducationViewModel` for presentation. When we wanted to present the education flow outside of the connection flow, we just create the `TapToPayEducationViewModel` directly and don't care about the payment event details, as it's not happening in a payment context. The event is the user tapping something else in the app, instead. 

The new design is more flexible, because `CardPresentPaymentEventDetails` is not a view model, it's just details about an event, and the presenter can decide what view model to make for each event and how to present it.

Unfortunately under the existing UIKit design, it's not so clean, as we have the same type of concrete view model class produced for every event, and it does need to be used to present a view. So unless we refactor the existing code to use the new approach, we can't reasonably include the education step in this way.

It would probably be ideal to do the refactor. But, it's also not factored in to the project so probably not a reasonable choice here. Since we're not doing it now, let's avoid making it harder to do it in the future. I think that having a separate presenter meets that need. We should try to keep it simple though!